samples/functions/csharp/simplechat/Functions.cs

```diff
@@ -90,7 +90,19 @@ public static IActionResult Home([HttpTrigger(AuthorizationLevel.Anonymous)] Htt
                 DataType = request.DataType
             });
 
-            return request.CreateResponse(BinaryData.FromString(new ClientContent("ack").ToString()), WebPubSubDataType.Json);
+            // retrieve counter from states.
+            var states = new ConnectionState(1);
+            var idle = 0.0;
+            if (connectionContext.States != null)
+            {
+                states = JsonConvert.DeserializeObject<ConnectionState>(connectionContext.States["connectionState"] as string);
+                idle = (DateTime.Now - states.Timestamp).TotalSeconds;
+                states.Update();
+            }
+            var response = request.CreateResponse(BinaryData.FromString(new ClientContent($"ack, idle: {idle}s, connection message counter: {states.Counter}").ToString()), WebPubSubDataType.Json);
+            response.SetState("connectionState", states);
```

Comment:
should be the serialized string for states

Reply:
updated.